### PR TITLE
corrected Vec2, Vec3 and Vec4 assignments

### PR DESCRIPTION
**Pull request by contriteobserver**
addresses issue #1923 which also cropped up in #1922

### CODE REVIEW

**Comment by contriteobserver** [comment]
added unit tests for RVec2, RVec3, RVec4

File: rajawali/src/main/java/org/rajawali3d/materials/shaders/AShaderBase.java [hunk 1]
```diff
@@ -294,31 +294,31 @@ public ShaderVar xy()
...
-			ShaderVar v = getReturnTypeForOperation(mDataType, mDataType);
+			ShaderVar v = new RFloat();
 			v.setName(this.mName + ".y");
 			v.mInitialized = true;
 			return v;
 		}
 		
 		public ShaderVar s()
 		{
-			ShaderVar v = getReturnTypeForOperation(mDataType, mDataType);
```
Comment (contriteobserver) [inline rajawali/src/main/java/org/rajawali3d/materials/shaders/AShaderBase.java]: I am confused, aside from float, what other data types would have a meaningful assign to x?

**Comment by contriteobserver** [comment]
added RVec2.assign(float, float) and RVec4.assign(float, float, float, float)

File: rajawali/src/main/java/org/rajawali3d/materials/shaders/AShaderBase.java (hunk 1 above)
Comment (jwoolston) [inline rajawali/src/main/java/org/rajawali3d/materials/shaders/AShaderBase.java]: something along the lines of `someVec4.x = 1` or `someVec4.y = 0xAB`

**Pull request merged**